tests/integrationv2/test_client_authentication.py

```diff
@@ -37,6 +37,25 @@ def assert_s2n_handshake_complete(results, protocol, provider, is_complete=True)
             expected_version)) not in results.stdout
 
 
+# If protocol version is TLS1.3 and lib-crypto is openssl-1.0.2, protocol version should downgrade to TLS1.2
+def verify_downgrade(protocol):
+    if protocol is Protocols.TLS13 and "openssl-1.0.2" in get_flag(S2N_PROVIDER_VERSION):
+        protocol_version = '33'
+    else:
+        protocol_version = protocol.value
+    return protocol_version
+
+
+def assert_s2n_handshake_complete_downgrade(results, protocol, is_complete=True):
```

Comment:
Given that these functions are only called once each, and are really small, I don't think they need to be functions at all. Their code can just go directly in the test.

Reply:
I created the function `verify_downgrade` since `protocol` isn't a parameter in the test. Checking with protocol is required to assert to actual downgrade.